modules/microshift-lvm-thin-volumes.adoc

```diff
@@ -0,0 +1,50 @@
+// Module included in the following assemblies:
+//
+// * microshift_storage/volume-snapshots-microshift.adoc
+
+:_content-type: CONCEPT
+[id="microshift-lvm-thin-volumes_{context}"]
+= About LVM thin volumes
+
+To use advanced storage features such as creating volume snapshots or volume cloning, you must provision a logical volume manager (LVM) thin-pool on the {op-system-ostree} host and configure both the logical volume manager storage (LVMS) provider and the cluster. LVM thin-pools must be attached to a volume group.
+
+[IMPORTANT]
+====
+To create CSI snapshots, you must configure thin volumes on the {op-system-ostree} host. The CSI does not support volume shrinking.
+====
+
+For LVMS to manage thin logical volumes (LVs), a thin-pool `device-class` must be specified in the `lvmd.yaml` file. Multiple thin-pool device classes are permitted.
+
+[NOTE]
+====
+To enable dynamic provisioning on a thin-pool, a `StorageClass` must be present on the cluster that specifies the source `device-class` in the `topolvm.io/device-class` parameter.
```

Comment:
This feels important, but I don't understand it. 

I think what this is trying to say is that the thin-pool needs to be associated with a StorageClass in order to be visible to the snapshotter. Is that all this is, @copejon , or is there something more subtle about it?

Reply:
Less than that even. This is just saying that if additional storage pools (exposed as device-classes) are configured, then additional storage classes must exist to expose the storage pools to users / workloads.  This being a doc for configuring thin vols, it seemed appropriate to include it as a reminder.